plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerRegistryAuthManager.java

```diff
@@ -0,0 +1,166 @@
+ *******************************************************************************/
+package org.eclipse.che.plugin.docker.client;
+
+import com.google.inject.Inject;
+
+import org.apache.commons.codec.binary.Base64;
+import org.eclipse.che.commons.annotation.Nullable;
+import org.eclipse.che.commons.json.JsonHelper;
+import org.eclipse.che.plugin.docker.client.dto.AuthConfig;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Class for preparing docker auth data.
+ *
+ * @author Mykola Morhun
+ */
+public class DockerRegistryAuthManager {
+
+    public static final String DEFAULT_REGISTRY = "";
+
+    private final InitialAuthConfig initialAuthConfig;
+
+    @Inject
+    public DockerRegistryAuthManager(InitialAuthConfig initialAuthConfig) {
+        this.initialAuthConfig = initialAuthConfig;
+    }
+
+    /**
+     * Looks for auth header for specified registry and encode it in base64.
+     * First searches in the params and then in the initial auth config.
+     * If nothing found empty encoded json will be returned.
+     *
+     * @param registry
+     *         registry to which API call will be applied
+     * @param paramsAuthConfig
+     *         auth data for current API call
+     * @return base64 encoded X-Registry-Auth header value
+     */
+    public String getXRegistryAuthHeaderValue(String registry, @Nullable Map<String,AuthConfig> paramsAuthConfig) {
+        if (DEFAULT_REGISTRY.equals(registry)) {
+            XRegistryAuthUnit auth = new XRegistryAuthUnit(initialAuthConfig.getDefaultUsername(), initialAuthConfig.getDefaultPassword());
+            return Base64.encodeBase64String(JsonHelper.toJson(auth).getBytes());
+        }
+
+        AuthConfig authConfig = null;
+        if (paramsAuthConfig != null) {
+            for(Map.Entry<String, AuthConfig> entry : paramsAuthConfig.entrySet()) {
+                AuthConfig value = entry.getValue();
+                if (value.getServeraddress().contains(registry)) {
+                    authConfig = value;
+                    break;
+                }
+            }
+        }
+
+        if (authConfig == null) {
+            for(Map.Entry<String, AuthConfig> entry : initialAuthConfig.getAuthConfigs().getConfigs().entrySet()) {
+                AuthConfig value = entry.getValue();
+                if (value.getServeraddress().contains(registry)) {
+                    authConfig = value;
+                    break;
+                }
+            }
+        }
+
+        if (authConfig != null) {
+            XRegistryAuthUnit auth = new XRegistryAuthUnit(authConfig.getUsername(), authConfig.getPassword());
+            return Base64.encodeBase64String(JsonHelper.toJson(auth).getBytes());
+        }
+
+        return Base64.encodeBase64String("{}".getBytes());
+    }
+
+    /**
+     * Looks for auth header for specified registry and encode it in base64.
+     * First searches in the params and then in the initial auth config.
```

Comment:
Replace `then in the initial auth config` with `then in the preconfigured system wide auth configs`